pkg/amphoracontrollers/flavors.go

```diff
@@ -0,0 +1,121 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amphoracontrollers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+	keystonev1 "github.com/openstack-k8s-operators/keystone-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/openstack"
+	octaviav1 "github.com/openstack-k8s-operators/octavia-operator/api/v1beta1"
+	"github.com/openstack-k8s-operators/octavia-operator/pkg/octavia"
+)
+
+func ensureNovaFlavors(osclient *openstack.OpenStack, log *logr.Logger) (string, error) {
+	client, err := octavia.GetComputeClient(osclient)
+	if err != nil {
+		return "", err
+	}
+
+	// Get Octavia flavors
+	listOpts := flavors.ListOpts{
+		AccessType: flavors.AllAccess,
+	}
+	allPages, err := flavors.ListDetail(client, listOpts).AllPages()
+	if err != nil {
+		return "", err
+	}
+	allFlavors, err := flavors.ExtractFlavors(allPages)
+	if err != nil {
+		return "", err
+	}
+	amphoraFlavors := make(map[string]flavors.Flavor)
+	for _, flavor := range allFlavors {
+		if strings.HasPrefix(flavor.Name, "octavia-") {
+			amphoraFlavors[flavor.Name] = flavor
+		}
+	}
+
+	isPublic := false
+	defaultFlavorsCreateOpts := []flavors.CreateOpts{
+		{
+			Name:       "octavia-amphora",
+			VCPUs:      1,
+			RAM:        1024,
+			Disk:       gophercloud.IntToPointer(3),
```

Comment:
As you pointed out in a discussion downsteam already, 3 GB might not be enough. I guess we can leave it like this for now, be it would probably be good to add at least a TODO comment to verify this is enough, or maybe we can make this configurable between a certain range >3.

Reply:
yes, the other solution would be to enable log offloading and disable local disk storage by default, in that case 3GB would be enough